figshare/Article.py

```diff
@@ -627,9 +642,18 @@ def find_matched_articles(self, articles):
                     # check curation folder for required files and setup data for further processing.
                     if (version_data is not None and len(version_data) > 0):
                         data = self.__check_curation_dir(version_data)
+                        version_no = "v" + str(data["version"]).zfill(2)
+                        i += 1
                         if (data["matched"] is True):
+                            total_file_size = version_data['size']
+                            self.total_all_articles_file_size += total_file_size
                             article_data[version_data['id']].append(data)
+                            no_matched += 1
+                            self.article_match_info[i] = f"article {data['id']} {version_no} ----- {data['author_dir']}"
+                        else:
+                            self.article_match_info[i] = f"article {data['id']} {version_no} ----- "
```

Comment:
In the console and logs, the list of matches is output after the matching process and space check. It took me an minute to process what was going on. Is there a particular reason for outputting it after the actual matching process and not immediately before L656? 

E.g.
```
2022-12-21 19:45:51,899:INFO:Log: Total matched article versions: 13.
2022-12-21 19:45:51,911:INFO:Log: Checking required space, script will stop if there's not enough space.
2022-12-21 19:45:51,935:INFO:Log: article 7873476 v01 ----- Jeff_Oliver_7873476
2022-12-21 19:45:51,936:INFO:Log: article 7873476 v02 ----- Jeff_Oliver_7873476
2022-12-21 19:45:51,936:INFO:Log: article 7873476 v03 ----- Jeff_Oliver_7873476
```

Reply:
@zoidy It seems to me that there's an issue with the logging not being thread safe, not the ordering. From my review, the comments should go like so due to how they're ordered in the code:

1. Finding matched articles.
2. article 7873476 v01 ----- Jeff_Oliver_7873476
article 7873476 v02 ----- Jeff_Oliver_7873476
article 7873476 v03 ----- Jeff_Oliver_7873476
3. Total matched article versions: {no_matched}.
4. Checking required space, script will stop if there's not enough space.

But this is how they're showing up for me (and this actually isn't consistent either):
Sun Jan  8 17:09:01 2023:INFO:Log - Finding matched articles.
Sun Jan  8 17:09:01 2023:INFO:Log - Total matched article versions: 12.
Sun Jan  8 17:09:01 2023:INFO:Log - Checking required space, script will stop if there's not enough space.
Sun Jan  8 17:09:01 2023:INFO:Log - article 7873476 v01 ----- Jeff_Oliver_7873476
Sun Jan  8 17:09:01 2023:INFO:Log - article 7873476 v02 ----- Jeff_Oliver_7873476

To resolve it we would need to implement threads in order to make the logging/terminal printing thread safe. @rubab please let us know if you have any other ideas of how we could resolve this.